src/tree/searchTree.ts

```diff
@@ -0,0 +1,94 @@
+import { observe } from 'mobx';
+import * as vscode from 'vscode';
+import { SEARCH_TRACKS_PLAYLIST_ID, Store } from '../store';
+import { getThemeIcon } from '../utils/iconUtils';
+import { getChildren } from './childrenLoader';
+import { AlbumTreeItem, PlayListTreeItem, TrackTreeItem } from './treeItems';
+import { ArtistTreeItem } from './treeItems/artistTreeItem';
+
+export class SearchTree implements vscode.TreeDataProvider<vscode.TreeItem> {
+    private _onDidChangeTreeData = new vscode.EventEmitter<vscode.TreeItem | undefined>();
+    readonly onDidChangeTreeData = this._onDidChangeTreeData.event;
+
+    constructor(private store: Store) {
+        observe(store, "searchResponse", () => {
+            this.refresh();
+        });
+    }
+
+    refresh(item?: vscode.TreeItem): void {
+        this._onDidChangeTreeData.fire(item);
+    }
+
+    getTreeItem(element: vscode.TreeItem): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        return element;
+    }
+
+    async getChildren(element?: vscode.TreeItem) {
```

Comment:
Need to refactor this method

Reply:
I think we should create issue for this refactoring or TODO marked?

Reply:
I've refactored this code, @AlexSnowLeo  could you look at this maybe you will have an idea how we can improve it and make even better.